src/s3.c

```diff
@@ -3688,5 +3705,15 @@ int main(int argc, char **argv)
         return -1;
     }
 
-    return 0;
+    if (failOnError && statusG != S3StatusOK) {
+        if (statusG < S3StatusErrorAccessDenied) {
```

Comment:
Why not just return -statusG, then you get every error code separate?  You can assign specific integers to the error codes in the enum to ensure that the same codes are always returned.